test/pico_time_test/pico_time_test.c

```diff
@@ -215,6 +216,35 @@ int main() {
     PICOTEST_CHECK(absolute_time_diff_us(near_the_end_of_time, at_the_end_of_time) > 0, "near the end of time should be before the end of time")
     PICOTEST_END_SECTION();
 
+    if (issue_195_test()) {
```

Comment:
Apologies if this is a stupid question, but this seems to be checking the return from `issue_195_test()` but AFAICT that function will only ever `return 0` ? Was that function supposed to `return 1` if the delta between `expected_count` and `issue_195_counter` was too big?

Reply:
nope; there are macros

Reply:
Perhaps I'm being _really_ stupid, but I can see that `issue_195_test()` has a `PICOTEST_START_SECTION` macro which [sets](https://github.com/raspberrypi/pico-sdk/blob/master/test/pico_test/include/pico/test.h#L22) `picotest_error_code = 0` and there's a `PICOTEST_END_SECTION` macro which [checks](https://github.com/raspberrypi/pico-sdk/blob/master/test/pico_test/include/pico/test.h#L24) `if (picotest_error_code != 0)`, but I can't tell where (in `issue_195_test()`) `picotest_error_code` actually gets modified :confused:

Reply:
the convention i am propagating is that tests return 0 on success or otherwise on failure which is what the macros do in general. Yes there is no actual check here atm (as it causes a hang in the failure case)... it _does not matter_

Reply:
> (as it causes a hang in the failure case)... it does not matter

Oh, that's the bit I was missing. Sorry for pestering you!